solr/solrj/src/java/org/apache/solr/client/solrj/impl/LBHttpSolrClient.java

```diff
@@ -192,7 +192,7 @@ protected SolrClient getClient(Endpoint endpoint) {
    */
   @Deprecated
   @Override
-  public String removeSolrServer(String server) {
+  public synchronized String removeSolrServer(String server) {
```

Comment:
[Q] The javadocs point to `removeSolrEndpoint` as the successor to this deprecated method: should that be `synchronized` as well?

(It probably doesn't matter terribly for branch_9x since removeSolrEndpoint currently calls removeSolrServer; mostly mentioning to make sure we remember to do it on the 'main' version of this PR)